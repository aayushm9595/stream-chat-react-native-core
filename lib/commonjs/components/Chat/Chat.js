var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Chat = void 0;

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _react = _interopRequireWildcard(require("react"));

var _reactNative = require("react-native");

var _dayjs = _interopRequireDefault(require("dayjs"));

var _useIsOnline2 = require("./hooks/useIsOnline");

var _useStreami18n = require("./hooks/useStreami18n");

var _ChatContext = require("../../contexts/chatContext/ChatContext");

var _ThemeContext = require("../../contexts/themeContext/ThemeContext");

var _TranslationContext = require("../../contexts/translationContext/TranslationContext");

var _package = require("../../../package.json");

var _jsxFileName = "/Users/vishalnarkhede/projects/projects/stream-chat-react-native/src/components/Chat/Chat.tsx";

var Chat = function Chat(props) {
  var children = props.children,
      client = props.client,
      i18nInstance = props.i18nInstance,
      _props$logger = props.logger,
      logger = _props$logger === void 0 ? function () {
    return null;
  } : _props$logger,
      style = props.style;

  var _useState = (0, _react.useState)(),
      _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
      channel = _useState2[0],
      setChannel = _useState2[1];

  var _useState3 = (0, _react.useState)({
    t: function t(key) {
      return key;
    },
    tDateTimeParser: function tDateTimeParser(input) {
      return (0, _dayjs["default"])(input);
    }
  }),
      _useState4 = (0, _slicedToArray2["default"])(_useState3, 2),
      translators = _useState4[0],
      setTranslators = _useState4[1];

  (0, _useStreami18n.useStreami18n)({
    i18nInstance: i18nInstance,
    setTranslators: setTranslators
  });

  var _useIsOnline = (0, _useIsOnline2.useIsOnline)(client),
      connectionRecovering = _useIsOnline.connectionRecovering,
      isOnline = _useIsOnline.isOnline;

  (0, _react.useEffect)(function () {
    client == null ? void 0 : client.setUserAgent("stream-chat-react-native-" + _reactNative.Platform.OS + "-" + _package.version);
    client.recoverStateOnReconnect = false;
  }, []);

  var setActiveChannel = function setActiveChannel(newChannel) {
    return setChannel(newChannel);
  };

  if (!translators.t) return null;
  var chatContext = {
    channel: channel,
    client: client,
    connectionRecovering: connectionRecovering,
    isOnline: isOnline,
    logger: logger,
    setActiveChannel: setActiveChannel
  };
  return _react["default"].createElement(_ChatContext.ChatProvider, {
    value: chatContext,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 176,
      columnNumber: 5
    }
  }, _react["default"].createElement(_TranslationContext.TranslationProvider, {
    value: translators,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 177,
      columnNumber: 7
    }
  }, _react["default"].createElement(_ThemeContext.ThemeProvider, {
    style: style,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 178,
      columnNumber: 9
    }
  }, children)));
};

exports.Chat = Chat;
//# sourceMappingURL=Chat.js.map