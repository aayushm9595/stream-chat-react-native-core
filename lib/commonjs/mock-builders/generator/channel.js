var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.generateChannel = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _uuid = require("uuid");

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

var generateChannel = function generateChannel() {
  var options = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {
    channel: {}
  };
  var optionsChannel = options.channel,
      optionsBesidesChannel = (0, _objectWithoutProperties2["default"])(options, ["channel"]);
  var id = optionsChannel && optionsChannel.id || (0, _uuid.v4)();
  var type = optionsChannel && optionsChannel.type || 'messaging';
  return _objectSpread({
    channel: _objectSpread({
      cid: type + ":" + id,
      config: {
        automod: 'disabled',
        automod_behavior: 'flag',
        commands: [{
          args: '[text]',
          description: 'Post a random gif to the channel',
          name: 'giphy',
          set: 'fun_set'
        }],
        connect_events: true,
        created_at: '2020-04-24T11:36:43.859020368Z',
        max_message_length: 5000,
        message_retention: 'infinite',
        mutes: true,
        name: 'messaging',
        reactions: true,
        read_events: true,
        replies: true,
        search: true,
        typing_events: true,
        updated_at: '2020-04-24T11:36:43.859022903Z',
        uploads: true,
        url_enrichment: true
      },
      created_at: '2020-04-28T11:20:48.578147Z',
      created_by: {
        banned: false,
        created_at: '2020-04-27T13:05:13.847572Z',
        id: 'vishal',
        last_active: '2020-04-28T11:21:08.353026Z',
        online: false,
        role: 'user',
        updated_at: '2020-04-28T11:21:08.357468Z'
      },
      frozen: false,
      id: id,
      type: type,
      updated_at: '2020-04-28T11:20:48.578147Z'
    }, optionsChannel),
    members: [],
    messages: []
  }, optionsBesidesChannel);
};

exports.generateChannel = generateChannel;
//# sourceMappingURL=channel.js.map