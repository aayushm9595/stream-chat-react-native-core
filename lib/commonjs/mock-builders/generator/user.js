var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.generateStaticUser = exports.generateUser = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _uuid = require("uuid");

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

var generateUser = function generateUser() {
  var options = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
  return _objectSpread({
    banned: false,
    created_at: '2020-04-27T13:39:49.331742Z',
    id: (0, _uuid.v4)(),
    image: (0, _uuid.v4)(),
    name: (0, _uuid.v4)(),
    online: false,
    role: 'user',
    updated_at: '2020-04-27T13:39:49.332087Z'
  }, options);
};

exports.generateUser = generateUser;

var generateStaticUser = function generateStaticUser(userNumber) {
  switch (userNumber) {
    case 0:
      return generateUser({
        id: 'DannyBoi1',
        image: 'https://images-na.ssl-images-amazon.com/images/I/51t29lLkg8L._AC_SL1000_.jpg',
        name: 'Dan'
      });

    case 1:
      return generateUser({
        id: 'GrantyBoi2',
        image: 'https://i.imgur.com/SLx06PP.png',
        name: 'Grant'
      });

    case 2:
      return generateUser({
        id: 'ViriBoi2',
        image: 'https://i.imgur.com/iNaC3K7.jpg',
        name: 'Vir'
      });

    default:
      return generateUser({
        id: 'NeiliBoi3',
        image: 'https://i.imgur.com/T68W8nR_d.webp?maxwidth=728&fidelity=grand',
        name: 'Neil'
      });
  }
};

exports.generateStaticUser = generateStaticUser;
//# sourceMappingURL=user.js.map