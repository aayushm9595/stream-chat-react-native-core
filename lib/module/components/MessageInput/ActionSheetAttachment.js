var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ActionSheetAttachment = void 0;

var _taggedTemplateLiteralLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteralLoose"));

var _react = _interopRequireDefault(require("react"));

var _reactNativeActionsheet = require("react-native-actionsheet");

var _IconSquare = require("../IconSquare");

var _TranslationContext = require("../../contexts/translationContext/TranslationContext");

var _styledComponents = require("../../styles/styledComponents");

var _jsxFileName = "/Users/vishalnarkhede/projects/projects/stream-chat-react-native/src/components/MessageInput/ActionSheetAttachment.tsx";

function _templateObject4() {
  var data = (0, _taggedTemplateLiteralLoose2["default"])(["\n  font-weight: bold;\n  ", ";\n"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = (0, _taggedTemplateLiteralLoose2["default"])(["\n  align-items: center;\n  flex-direction: row;\n  height: 100%;\n  justify-content: space-between;\n  padding-horizontal: 20px;\n  width: 100%;\n  ", ";\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = (0, _taggedTemplateLiteralLoose2["default"])(["\n  ", ";\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = (0, _taggedTemplateLiteralLoose2["default"])(["\n  align-items: center;\n  flex-direction: row;\n  justify-content: flex-start;\n  padding-left: 20px;\n  width: 100%;\n  ", ";\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var iconGallery = require("../../images/icons/icon_attach-media.png");

var iconClose = require("../../images/icons/icon_close.png");

var iconFolder = require("../../images/icons/icon_folder.png");

var ActionSheetButtonContainer = _styledComponents.styled.View(_templateObject(), function (_ref) {
  var theme = _ref.theme;
  return theme.messageInput.actionSheet.buttonContainer.css;
});

var ActionSheetButtonText = _styledComponents.styled.Text(_templateObject2(), function (_ref2) {
  var theme = _ref2.theme;
  return theme.messageInput.actionSheet.buttonText.css;
});

var ActionSheetTitleContainer = _styledComponents.styled.View(_templateObject3(), function (_ref3) {
  var theme = _ref3.theme;
  return theme.messageInput.actionSheet.titleContainer.css;
});

var ActionSheetTitleText = _styledComponents.styled.Text(_templateObject4(), function (_ref4) {
  var theme = _ref4.theme;
  return theme.messageInput.actionSheet.titleText.css;
});

var AttachmentActionSheetItem = function AttachmentActionSheetItem(props) {
  var icon = props.icon,
      key = props.key,
      testID = props.testID,
      text = props.text;
  return _react["default"].createElement(ActionSheetButtonContainer, {
    key: key,
    testID: testID,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 61,
      columnNumber: 5
    }
  }, _react["default"].createElement(_IconSquare.IconSquare, {
    icon: icon,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 62,
      columnNumber: 7
    }
  }), _react["default"].createElement(ActionSheetButtonText, {
    __source: {
      fileName: _jsxFileName,
      lineNumber: 63,
      columnNumber: 7
    }
  }, text));
};

var ActionSheetAttachment = function ActionSheetAttachment(props) {
  var closeAttachActionSheet = props.closeAttachActionSheet,
      pickFile = props.pickFile,
      pickImage = props.pickImage,
      setAttachActionSheetRef = props.setAttachActionSheetRef,
      styles = props.styles;

  var _useTranslationContex = (0, _TranslationContext.useTranslationContext)(),
      t = _useTranslationContex.t;

  return _react["default"].createElement(_reactNativeActionsheet.ActionSheetCustom, {
    onPress: function onPress(index) {
      setTimeout(function () {
        switch (index) {
          case 0:
            pickImage();
            break;

          case 1:
            pickFile();
            break;

          default:
        }
      }, 201);
    },
    options: [_react["default"].createElement(AttachmentActionSheetItem, {
      icon: iconGallery,
      key: "upload-photo-item",
      testID: "upload-photo-item",
      text: t('Upload a photo'),
      __source: {
        fileName: _jsxFileName,
        lineNumber: 117,
        columnNumber: 9
      }
    }), _react["default"].createElement(AttachmentActionSheetItem, {
      icon: iconFolder,
      key: "upload-file-item",
      testID: "upload-file-item",
      text: t('Upload a file'),
      __source: {
        fileName: _jsxFileName,
        lineNumber: 123,
        columnNumber: 9
      }
    })],
    ref: setAttachActionSheetRef,
    styles: styles,
    title: _react["default"].createElement(ActionSheetTitleContainer, {
      __source: {
        fileName: _jsxFileName,
        lineNumber: 133,
        columnNumber: 9
      }
    }, _react["default"].createElement(ActionSheetTitleText, {
      __source: {
        fileName: _jsxFileName,
        lineNumber: 134,
        columnNumber: 11
      }
    }, t('Add a file')), _react["default"].createElement(_IconSquare.IconSquare, {
      icon: iconClose,
      onPress: closeAttachActionSheet,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 135,
        columnNumber: 11
      }
    })),
    __source: {
      fileName: _jsxFileName,
      lineNumber: 101,
      columnNumber: 5
    }
  });
};

exports.ActionSheetAttachment = ActionSheetAttachment;
//# sourceMappingURL=ActionSheetAttachment.js.map