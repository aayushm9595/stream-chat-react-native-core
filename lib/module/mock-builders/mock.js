var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getRandomInt = exports.getTestClientWithUser = exports.getTestClient = void 0;

var _regenerator = _interopRequireDefault(require("@babel/runtime/regenerator"));

var _asyncToGenerator2 = _interopRequireDefault(require("@babel/runtime/helpers/asyncToGenerator"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _streamChat = require("stream-chat");

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

var apiKey = 'API_KEY';
var token = 'dummy_token';

var setUser = function setUser(client, user) {
  return new Promise(function (resolve) {
    client.connectionId = 'dumm_connection_id';
    client.user = user;
    client.user.mutes = [];
    client._user = _objectSpread({}, user);
    client.userID = user.id;
    client.userToken = token;
    resolve();
  });
};

function mockClient(client) {
  jest.spyOn(client, '_setToken').mockImplementation();
  jest.spyOn(client, '_setupConnection').mockImplementation();
  client.tokenManager = {
    getToken: jest.fn(function () {
      return token;
    }),
    tokenReady: jest.fn(function () {
      return true;
    })
  };
  client.setUser = setUser.bind(null, client);
  return client;
}

var getTestClient = function getTestClient() {
  return mockClient(new _streamChat.StreamChat(apiKey));
};

exports.getTestClient = getTestClient;

var getTestClientWithUser = function () {
  var _ref = (0, _asyncToGenerator2["default"])(_regenerator["default"].mark(function _callee(user) {
    var client;
    return _regenerator["default"].wrap(function _callee$(_context) {
      while (1) {
        switch (_context.prev = _context.next) {
          case 0:
            client = mockClient(new _streamChat.StreamChat(apiKey));
            _context.next = 3;
            return setUser(client, user);

          case 3:
            client.wsPromise = Promise.resolve();
            return _context.abrupt("return", client);

          case 5:
          case "end":
            return _context.stop();
        }
      }
    }, _callee);
  }));

  return function getTestClientWithUser(_x) {
    return _ref.apply(this, arguments);
  };
}();

exports.getTestClientWithUser = getTestClientWithUser;

var getRandomInt = function getRandomInt(min, max) {
  min = Math.ceil(min);
  max = Math.floor(max);
  return Math.floor(Math.random() * (max - min)) + min;
};

exports.getRandomInt = getRandomInt;
//# sourceMappingURL=mock.js.map