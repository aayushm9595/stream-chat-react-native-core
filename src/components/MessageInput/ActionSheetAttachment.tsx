import React from 'react';
import { ActionSheetCustom } from 'react-native-actionsheet';

import type {
  ImageRequireSource,
  StyleProp,
  TextStyle,
  ViewStyle,
} from 'react-native';

import { IconSquare } from '../IconSquare';

import { useTranslationContext } from '../../contexts/translationContext/TranslationContext';

import { styled } from '../../styles/styledComponents';

const iconGallery: ImageRequireSource = require('../../images/icons/icon_attach-media.png');
const iconClose: ImageRequireSource = require('../../images/icons/icon_close.png');
const iconFolder: ImageRequireSource = require('../../images/icons/icon_folder.png');

const ActionSheetButtonContainer = styled.View`
  align-items: center;
  flex-direction: row;
  justify-content: flex-start;
  padding-left: 20px;
  width: 100%;
  ${({ theme }) => theme.messageInput.actionSheet.buttonContainer.css};
`;

const ActionSheetButtonText = styled.Text`
  ${({ theme }) => theme.messageInput.actionSheet.buttonText.css};
`;

const ActionSheetTitleContainer = styled.View`
  align-items: center;
  flex-direction: row;
  height: 100%;
  justify-content: space-between;
  padding-horizontal: 20px;
  width: 100%;
  ${({ theme }) => theme.messageInput.actionSheet.titleContainer.css};
`;

const ActionSheetTitleText = styled.Text`
  font-weight: bold;
  ${({ theme }) => theme.messageInput.actionSheet.titleText.css};
`;

type AttachmentActionSheetItemProps = {
  icon: ImageRequireSource;
  key: string | number;
  text: string;
  testID?: string;
};

const AttachmentActionSheetItem: React.FC<AttachmentActionSheetItemProps> = (
  props,
) => {
  const { icon, key, testID, text } = props;
  return (
    <ActionSheetButtonContainer key={key} testID={testID}>
      <IconSquare icon={icon} />
      <ActionSheetButtonText>{text}</ActionSheetButtonText>
    </ActionSheetButtonContainer>
  );
};

export type ActionSheetStyles = {
  body?: StyleProp<ViewStyle>;
  buttonBox?: StyleProp<ViewStyle>;
  buttonText?: StyleProp<TextStyle>;
  cancelButtonBox?: StyleProp<ViewStyle>;
  messageBox?: StyleProp<ViewStyle>;
  messageText?: StyleProp<TextStyle>;
  overlay?: StyleProp<TextStyle>;
  titleBox?: StyleProp<ViewStyle>;
  titleText?: StyleProp<TextStyle>;
  wrapper?: StyleProp<ViewStyle>;
};

export type ActionSheetProps = {
  closeAttachActionSheet: () => void;
  pickFile: () => Promise<void>;
  pickImage: () => Promise<void>;
  setAttachActionSheetRef: (ref: ActionSheetCustom | null) => void;
  styles?: ActionSheetStyles;
};

export const ActionSheetAttachment: React.FC<ActionSheetProps> = (props) => {
  const {
    closeAttachActionSheet,
    pickFile,
    pickImage,
    setAttachActionSheetRef,
    styles,
  } = props;

  const { t } = useTranslationContext();

  return (
    <ActionSheetCustom
      onPress={(index) => {
        // https://github.com/beefe/react-native-actionsheet/issues/36
        setTimeout(() => {
          switch (index) {
            case 0:
              pickImage();
              break;
            case 1:
              pickFile();
              break;
            default:
          }
        }, 201); // 201ms to fire after the animation is complete https://github.com/beefe/react-native-actionsheet/blob/master/lib/ActionSheetCustom.js#L78
      }}
      options={[
        <AttachmentActionSheetItem
          icon={iconGallery}
          key='upload-photo-item'
          testID='upload-photo-item'
          text={t('Upload a photo')}
        />,
        <AttachmentActionSheetItem
          icon={iconFolder}
          key='upload-file-item'
          testID='upload-file-item'
          text={t('Upload a file')}
        />,
      ]}
      ref={setAttachActionSheetRef}
      styles={styles}
      title={
        <ActionSheetTitleContainer>
          <ActionSheetTitleText>{t('Add a file')}</ActionSheetTitleText>
          <IconSquare icon={iconClose} onPress={closeAttachActionSheet} />
        </ActionSheetTitleContainer>
      }
    />
  );
};
