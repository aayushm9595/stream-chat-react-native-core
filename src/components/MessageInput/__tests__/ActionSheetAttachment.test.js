import React from 'react';
import { fireEvent, render, waitFor } from '@testing-library/react-native';

import { getOrCreateChannelApi } from 'mock-builders/api/getOrCreateChannel';
import { useMockedApis } from 'mock-builders/api/useMockedApis';
import { generateChannel } from 'mock-builders/generator/channel';
import { generateUser } from 'mock-builders/generator/user';
import { getTestClientWithUser } from 'mock-builders/mock';

import { ActionSheetAttachment } from '../ActionSheetAttachment';

import { Chat } from '../../Chat/Chat';

describe('ActionSheetAttachment', () => {
  const clientUser = generateUser();
  let chatClient;
  let channel;

  const getComponent = (props = {}) => (
    <Chat client={chatClient}>
      <ActionSheetAttachment {...props} />
    </Chat>
  );

  const initializeChannel = async (c) => {
    useMockedApis(chatClient, [getOrCreateChannelApi(c)]);

    channel = chatClient.channel('messaging');

    await channel.watch();
  };

  beforeEach(async () => {
    chatClient = await getTestClientWithUser(clientUser);
  });

  afterEach(() => {
    channel = null;
  });

  it('should render ActionSheetAttachment and click through items and close icon', async () => {
    const props = {
      closeAttachActionSheet: jest.fn(),
      pickFile: jest.fn(),
      pickImage: jest.fn(),
    };

    await initializeChannel(generateChannel());

    const {
      getAllByTestId,
      getByTestId,
      queryAllByTestId,
      queryByTestId,
      queryByText,
      toJSON,
    } = render(getComponent(props));

    await waitFor(() => {
      expect(queryByTestId('upload-photo-item')).toBeTruthy();
      expect(queryByTestId('upload-file-item')).toBeTruthy();
      expect(queryAllByTestId('icon-square')).toBeTruthy();
      expect(queryByText('Add a file')).toBeTruthy();
      expect(props.closeAttachActionSheet).toHaveBeenCalledTimes(0);
      expect(props.pickFile).toHaveBeenCalledTimes(0);
      expect(props.pickImage).toHaveBeenCalledTimes(0);
    });

    fireEvent.press(getByTestId('upload-photo-item'));

    await waitFor(() => {
      expect(props.closeAttachActionSheet).toHaveBeenCalledTimes(0);
      expect(props.pickFile).toHaveBeenCalledTimes(0);
      expect(props.pickImage).toHaveBeenCalledTimes(1);
    });

    fireEvent.press(getByTestId('upload-file-item'));

    await waitFor(() => {
      expect(props.closeAttachActionSheet).toHaveBeenCalledTimes(0);
      expect(props.pickFile).toHaveBeenCalledTimes(1);
      expect(props.pickImage).toHaveBeenCalledTimes(1);
    });

    fireEvent.press(getAllByTestId('icon-square')[0]);

    await waitFor(() => {
      expect(props.closeAttachActionSheet).toHaveBeenCalledTimes(1);
      expect(props.pickFile).toHaveBeenCalledTimes(1);
      expect(props.pickImage).toHaveBeenCalledTimes(1);
    });

    const snapshot = toJSON();

    await waitFor(() => {
      expect(snapshot).toMatchSnapshot();
    });
  });
});
