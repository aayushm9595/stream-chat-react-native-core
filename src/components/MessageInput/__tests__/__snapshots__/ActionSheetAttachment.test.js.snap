// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionSheetAttachment should render ActionSheetAttachment and click through items and close icon 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={true}
  visible={false}
>
  <View
    style={
      Array [
        Array [
          Object {
            "flex": 1,
            "flexDirection": "row",
          },
        ],
      ]
    }
  >
    <Text
      onPress={[Function]}
      style={
        Array [
          Array [
            Object {
              "backgroundColor": "#000",
              "bottom": 0,
              "left": 0,
              "opacity": 0.4,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
          ],
        ]
      }
    />
    <View
      style={
        Object {
          "alignSelf": "flex-end",
          "backgroundColor": "#e5e5e5",
          "flex": 1,
          "height": 141,
          "transform": Array [
            Object {
              "translateY": 141,
            },
          ],
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "height": 40,
              "justifyContent": "center",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": "100%",
                "justifyContent": "space-between",
                "paddingHorizontal": 20,
                "width": "100%",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontWeight": "bold",
                },
              ]
            }
          >
            Add a file
          </Text>
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.05)",
                "borderRadius": 5,
                "opacity": 1,
                "paddingBottom": 5,
                "paddingLeft": 5,
                "paddingRight": 5,
                "paddingTop": 5,
              }
            }
            testID="icon-square"
          >
            <Image
              source={
                Object {
                  "testUri": "../../../src/images/icons/icon_close.png",
                }
              }
              style={
                Object {
                  "height": 15,
                  "width": 15,
                }
              }
            />
          </View>
        </View>
      </View>
      <RCTScrollView
        scrollEnabled={false}
      >
        <View>
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fff",
                  "height": 50,
                  "justifyContent": "center",
                  "marginTop": 0.5,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                    "paddingLeft": 20,
                    "width": "100%",
                  },
                ]
              }
              testID="upload-photo-item"
            >
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "rgba(0, 0, 0, 0.05)",
                      "borderRadius": 5,
                      "paddingBottom": 5,
                      "paddingLeft": 5,
                      "paddingRight": 5,
                      "paddingTop": 5,
                    },
                  ]
                }
                testID="icon-square"
              >
                <Image
                  source={
                    Object {
                      "testUri": "../../../src/images/icons/icon_attach-media.png",
                    }
                  }
                  style={
                    Object {
                      "height": 15,
                      "width": 15,
                    }
                  }
                />
              </View>
              <Text
                style={
                  Array [
                    Object {},
                  ]
                }
              >
                Upload a photo
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fff",
                  "height": 50,
                  "justifyContent": "center",
                  "marginTop": 0.5,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                    "paddingLeft": 20,
                    "width": "100%",
                  },
                ]
              }
              testID="upload-file-item"
            >
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "rgba(0, 0, 0, 0.05)",
                      "borderRadius": 5,
                      "paddingBottom": 5,
                      "paddingLeft": 5,
                      "paddingRight": 5,
                      "paddingTop": 5,
                    },
                  ]
                }
                testID="icon-square"
              >
                <Image
                  source={
                    Object {
                      "testUri": "../../../src/images/icons/icon_folder.png",
                    }
                  }
                  style={
                    Object {
                      "height": 15,
                      "width": 15,
                    }
                  }
                />
              </View>
              <Text
                style={
                  Array [
                    Object {},
                  ]
                }
              >
                Upload a file
              </Text>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
</Modal>
`;
